packages/block-library/src/comments-pagination/index.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Server-side rendering of the `core/comments-pagination` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/comments-pagination` block on the server.
+ *
+ * @param array  $attributes Block attributes.
+ * @param string $content    Block default content.
+ *
+ * @return string Returns the wrapper for the Comments pagination.
+ */
+function render_block_core_comments_pagination( $attributes, $content ) {
+	if ( empty( trim( $content ) ) ) {
```

Comment:
That’s a bit surprising but I checked the history of the Query Pagination block and it looks like they had an issue with rendering of an empty `div` when there was no pagination to show. Basically it bails out early when inner blocks render nothing.